ast/check.go

```diff
@@ -146,7 +146,97 @@ func (tc *typeChecker) checkLanguageBuiltins(env *TypeEnv, builtins map[string]*
 	return env
 }
 
+func getPrefixToOverride(name string, env *TypeEnv, rule *Rule) (string, types.Type, *Error) {
```

Comment:
Having a look at this again, I'm not sure the recursion is even warranted if taking this route:

```go
func getPrefixToOverridefRef(ref Ref, env *TypeEnv, rule *Rule) (Ref, types.Type) {
	for i := 1; i < len(ref); i++ {
		t := env.tree.Get(ref[:i])
		if t != nil {
			return ref[:i], t
		}
	}
	return nil, nil
}
```
or some such thing...? This might even make sense as an addition to the `TypeEnv` API...? 🤔

Reply:
Thanks! Added to TypeEnv.
Done (commit https://github.com/open-policy-agent/opa/pull/3123/commits/3eb2e9f68305c617c8020dc8a984d36cca2be2e4)